server/app/mutations/grid_services/common.rb

```diff
@@ -84,6 +86,30 @@ def build_grid_service_secrets(existing_secrets)
       service_secrets
     end
 
+    def build_service_volumes()
+      service_volumes = []
+      self.volumes.each do |vol|
+        vol_spec = parse_volume(vol)
+        if vol_spec[:volume]
```

Comment:
then it just creates the `ServiceVolume` with the other args that has been parsed from the volume string. So this `if`block only "replaces" the parsed named volume string with the actual volume reference.